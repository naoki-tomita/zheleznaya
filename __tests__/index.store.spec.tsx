import { h, render, createStore, Component } from "../index";

describe("#createStore", () => {
  it("should rerender when store changed.", () => {
    const store = createStore({ count: 0 });
    const App: Component = () => {
      return (
        <div>{store.count}</div>
      );
    }
    render(<App />);
    expect(document.body).toMatchSnapshot();

    store.count = 2;
    expect(document.body).toMatchSnapshot();
  });
});
