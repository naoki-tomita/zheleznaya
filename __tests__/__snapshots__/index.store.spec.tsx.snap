// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createStore should rerender when store changed. 1`] = `
<body>
  <div>
    <div>
      0
    </div>
  </div>
</body>
`;

exports[`#createStore should rerender when store changed. 2`] = `
<body>
  <div>
    <div>
      2
    </div>
  </div>
</body>
`;
